### DISCUSSION TITLE
Plot predictions throwing  error for the CPU and CUDA type

### DISCUSSION THREAD

**Discussion by Heyymant** [mrdbourke/pytorch-deep-learning#751] [Q&A]
![image_2023-12-09_200720772](https://github.com/mrdbourke/pytorch-deep-learning/assets/76215033/7dd6c792-6b79-4d5b-8583-b9405a800db1)

I am getting the same error even after changing the device settings, can someone help me to rectify this error, (yea, I have tried restarting the notebook as well)

**Comment by sametucuncu** [comment]
I think you do not need to put .cpu() becuase they are not torch tensors. But if you share the whole code, I might help better.

**Reply by shisirkha** [reply]
Yeah, if he shares the entire we can help him.

**Comment by shisirkha** [comment]
Got ya first convert the list of numpy array into a torch.Tensor type like this then try.
example:
```
train_data = torch.from_numpy(train_data).cpu()
```
or do this:
```
train_data= torch.from_numpy(X_train).cpu()
train_label = torch.from_numpy(y_train).cpu() 
test_data = torch.from_numpy(X_test).cpu()
test_label = torch.from_numpy(y_train).cpu()

#then do this

plot_predictions(train_data,train_label,test_data,test_label)
```
Or if you want your variable name to be same do this:
```
X_train= torch.from_numpy(X_train).cpu()
y_train = torch.from_numpy(y_train).cpu() 
X_test = torch.from_numpy(X_test).cpu()
y_test = torch.from_numpy(y_train).cpu()

#then do this

plot_predictions(X_train,y_train,X_test,y_test)
```
use this example to define all your numpy array to a tensor.
I hope this helps
Good luck self learner
